opennms-config/src/test/java/org/opennms/netmgt/config/PollerConfigReloadIT.java

```diff
@@ -131,4 +141,81 @@ private InputStream setIncludeUrlFileInConfig(URL configUrl) throws IOException
         String modifiedString = marshalledString.replace("${INCLUDE_URL_FILE}", includeUrlFile.getAbsolutePath());
         return IOUtils.toInputStream(modifiedString, Charset.defaultCharset());
     }
+
+    @Test
+    public void testPollerConfigReloadFail() throws Exception {
+        final File temporaryFile = File.createTempFile("poller-configuration-", ".xml", new File(PollerConfigReloadIT.class.getResource("/etc").getFile()));
+        PollerConfigFactory.setPollerConfigFile(temporaryFile);
+
+        final AtomicBoolean invalid = new AtomicBoolean(false);
+        FilterDaoFactory.setInstance(new FilterDao() {
+            @Override
+            public SortedMap<Integer, String> getNodeMap(String rule) throws FilterParseException {
+                return null;
+            }
+
+            @Override
+            public Map<InetAddress, Set<String>> getIPAddressServiceMap(String rule) throws FilterParseException {
+                return null;
+            }
+
+            @Override
+            public void flushActiveIpAddressListCache() {
+            }
+
+            @Override
+            public List<InetAddress> getIPAddressList(String rule) throws FilterParseException {
+                return null;
+            }
+
+            @Override
+            public boolean isValid(String addr, String rule) throws FilterParseException {
+                return false;
+            }
+
+            @Override
+            public boolean isRuleMatching(String rule) throws FilterParseException {
+                return false;
+            }
+
+            @Override
+            public void validateRule(String rule) throws FilterParseException {
+                if (invalid.get()) {
+                    throw new FilterParseException("Something fishy");
+                }
+            }
+
+            @Override
+            public List<InetAddress> getActiveIPAddressList(String rule) throws FilterParseException {
+                return Lists.newArrayList();
+            }
+        });
```

Comment:
Hint: You could simplify this by using `EasyMock.createNiceMock(FilterDao.class)` instead and just mock the `validateRule`method.